feat(shared-data, api): Add evo tips definition and restrictions on loading

Laura-Danielle opened the pull request:
# Overview

Add evo tips definition but as a labware so that liquid handling actions can happen in the 'labware'.

## Test Plan and Hands on Testing

Need to check that the stackup works as expected

## Changelog

- Add evo tips definition
- Add stacking component to nest deepwell plate
- Add error check for the evo tips definition so that it is not accidentally loaded onto the deck

## Review requests

Check that the labware role makes sense and the stacking part.

## Risk assessment

Low. Adding a new labware.

Laura-Danielle committed:
feat(shared_data): Add labware definitions for evotips

Laura-Danielle committed:
add validation and the definitions to the latest labware def list

CaseyBatten commented on shared-data/js/__tests__/labwareDefQuirks.test.ts:
```diff
@@ -14,6 +14,7 @@ const EXPECTED_VALID_QUIRKS = [
   'gripperIncompatible',
   'tiprackAdapterFor96Channel',
   'stackingMaxFive',
```
This stackingMaxFive quirk can go away now that its unused in place of the stackLimit and compatibleParentLabware fields.

CaseyBatten commented on shared-data/labware/definitions/2/evotips_opentrons_96_labware/1.json:
```diff
@@ -0,0 +1,1032 @@
+	"namespace": "opentrons",
+	"allowedRoles": ["labware"],
+	"version": 1,
+	"stackLimit": 5,
+	"compatibleParentLabware": ["nest_96_wellplate_2ml_deep", "nest_1_reservoir_195ml"],
```
Looks good, are these the only reservoirs we plan on supporting for now?

CaseyBatten commented on api/src/opentrons/protocol_engine/resources/labware_validation.py:
```diff
@@ -44,6 +44,14 @@ def validate_labware_can_be_stacked(
     return below_labware_load_name in top_labware_definition.stackingOffsetWithLabware
 
 
+def validate_labware_can_be_ondeck(definition: LabwareDefinition) -> bool:
+    """Validate that the labware being loaded onto the deck can sit in a slot."""
+    return (
+        definition.parameters.quirks is None
+        or "stackingOnly" not in definition.parameters.quirks
+    )
+
+
```
This quirk is fine, however theres some parallel work going on with the lids behavior that introduced a new labware parameter called `isDeckSlotCompatible` that could be used for the same thing as this quirk. We'll be using it for the tiprack lids and any other lids that cannot be placed on deck, and in theory could use it here too, although it would need to be updated to apply to regular labware loads as well and not just lid stacks. Not a blocker but eventually we'll probably want to consolidate the two.

CaseyBatten commented on shared-data/labware/definitions/2/evotips_opentrons_96_labware/1.json:
```diff
@@ -0,0 +1,1032 @@
+	  "loadName": "evotips_opentrons_96_labware"
+	},
+	"namespace": "opentrons",
+	"allowedRoles": ["labware"],
```
Of note currently this will raise an error when we build a stackup of:
96ch Evo Tips Adapter -> Reservoir -> Evotips Labware due to a raise condition occurring in `protocol_engine/state/labware.py` which enforces that hybrid stacks must consist of at most an adapter-labware-lid combination, rather than an adapter-labware-labware combination. Given the new rules we have around stackLimits and compatibleParentLabware validations, I think we can do away with that raise case and allow this to be considered a _labware_ and not a _lid_.

CaseyBatten commented on shared-data/labware/definitions/2/evotips_flex_96_tiprack_adapter/1.json:
```diff
@@ -0,0 +1,41 @@
+	],
+	"parameters": {
+	  "format": "96Standard",
+	  "quirks": ["tiprackAdapterFor96Channel", "stackingMaxFive"],
```
stackingMaxFive can be removed here, do we need to add a stackLimit to this labware? It's at the "bottom" of the stack so I think not?

CaseyBatten reviewed (approved):
Overall the labware definition looks good, we'll need to make final determinations on how its used (as a labware or as a "lid") before merging and also on how publicly it will be available, and how it will impact LPC.

CaseyBatten committed:
addition of evotips adapter as valid parent for evotips and on deck raise fix

TODO determine stacking overlap for tiprack adapter with evo tips

CaseyBatten committed:
update labware definition relative to LPC offsets

CaseyBatten committed:
linting fixes

CaseyBatten committed:
tip length and overlap

CaseyBatten committed:
addition of gripper values and correction of tests

CaseyBatten committed:
migrate definitions from v2 to v3

CaseyBatten committed:
labware source update

github-actions[bot] committed:
fix(analyses-snapshot-testing): heal PLAT-350-add-evo-tips-tiprack-definitions snapshots (#17361)

This PR was requested on the PR https://github.com/Opentrons/opentrons/pull/17237

CaseyBatten committed:
Merge branch 'chore_release-8.3.0' into PLAT-350-add-evo-tips-tiprack-definitions
